Team-C/test_wizard/test/widget_test.dart

```diff
@@ -5,26 +5,26 @@
 // gestures. You can also use WidgetTester to find child widgets in the widget
 // tree, read text, and verify that the values of widget properties are correct.
 
-import 'package:flutter/material.dart';
+
 import 'package:flutter_test/flutter_test.dart';
 
 import 'package:test_wizard/main.dart';
 
 void main() {
-  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
+  testWidgets('Login Screen', (WidgetTester tester) async {
     // Build our app and trigger a frame.
     await tester.pumpWidget(const MyApp());
 
     // Verify that our counter starts at 0.
-    expect(find.text('0'), findsOneWidget);
-    expect(find.text('1'), findsNothing);
+    expect(find.text('Login'), findsOneWidget);
+   // expect(find.text('1'), findsNothing);
 
     // Tap the '+' icon and trigger a frame.
-    await tester.tap(find.byIcon(Icons.add));
-    await tester.pump();
+    //await tester.tap(find.byIcon(Icons.login));
+    //await tester.pump();
 
     // Verify that our counter has incremented.
-    expect(find.text('0'), findsNothing);
-    expect(find.text('1'), findsOneWidget);
+    //expect(find.text('0'), findsNothing);
+   //expect(find.text('1'), findsOneWidget);
   });
```

Comment:
I'll have to come back and look at this later, but good start on this, thanks Lana!

Reply:
Ok @Lanaismail I finally figured out what is happening: When you call `tester.pumpWidget(const MyApp())` you are starting the app from the beginning, which means the `LoginScreen` widget is not yet on the page - in the routes you have the `'/'` route set to an if statement based on whether the `ConnectionState` is `waiting`.  So, the test is correct: at the point you are testing from this file, there is only a `CircularProgressIndicator` on the page, nothing from `LoginScreen` or `HomeScreen` is yet displaying. The way to get around this is below:
```
await tester.pumpWidget(const MaterialApp(home: LoginScreen()));
    expect(find.text("Login"), findsExactly(2));
```
We need `MaterialApp` to set some defaults for the `LoginScreen`, you can't test a component with `Scaffold` and `AppBar` in it like a normal stand-alone widget. `findsExactly(2)` because "Login" is in the AppBar and the button. 

Sorry this took so long, it took me a while to figure out! Hopefuly it helps, and let me know if you need more info.